### ISSUE TITLE
False positive error "incompatible type" on bound TypeVar

### ISSUE THREAD

**Issue by epenet** [python/mypy#13596]
**Bug Report**
For this code:
```python
from typing import Generic, TypeVar, Union

ListT = TypeVar("ListT", str, float)


class TestListA(Generic[ListT]):
    def __init__(self, device: ListT) -> None:
        self._device: ListT = device


class TestListB(TestListA[ListT]):
    def __init__(self, device: ListT) -> None:
        super().__init__(device)
```

mypy outputs error:
```console
test.py:13: error: Argument 1 to "__init__" of "TestListA" has incompatible type "str"; expected "ListT"  [arg-type]
test.py:13: error: Argument 1 to "__init__" of "TestListA" has incompatible type "float"; expected "ListT"  [arg-type]
```

which is false positive.

**Your Environment**

- Mypy version used: `mypy 0.971 (compiled: yes)`
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: `Python 3.9.12`
- Operating system and version:

**Comment by epenet** [comment]
In a similar way, I have a false positive on `[assignment]`:

```python
from typing import Generic, TypeVar

ListT = TypeVar("ListT", str, float)


class TestListA(Generic[ListT]):
    def __setitem__(self, key: str, entry: ListT) -> None:
        pass


class TestListB(TestListA[ListT]):
    def __setitem__(self, key: str, entry: ListT) -> None:
        super().__setitem__(key, entry)
```

which incorrectly reports
```console
test.py:13: error: Incompatible types in assignment (expression has type "str", target has type "ListT")  [assignment]
test.py:13: error: Incompatible types in assignment (expression has type "float", target has type "ListT")  [assignment]
```
